scene/3d/skeleton.cpp

```diff
@@ -257,7 +257,11 @@ void Skeleton::_notification(int p_what) {
 				rest_global_inverse_dirty = false;
 			}
 
-			Transform global_transform = get_global_transform();
+			Transform global_transform;
+			// added in case the skeleton has exited the tree while an animation was still running
+			// and the message was still passed through. In other words, the skeleton may still have an
+			// animation running on a node while it is instanced but not within the tree.
+			if (is_inside_tree()) global_transform = get_global_transform();
```

Comment:
That seems hacky to me, why not do
```
if (!is_inside_tree())
	return;
```
at the top of the `case` instead?

Reply:
well less side effect since the behavior isn't changed at all, it just skips the error message. But really we should be concerned why NOTIFICATION_UPDATE_SKELETON is being called when it's outside the tree... which is my pejorative I guess. How about I just keep this change in my internal fork for now, and when I'm more confident  with the solution I'll re-open this pull request.